File: Python_Problems/Drawing_Book.py

#
# Complete the 'pageCount' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER p
#

def pageCount(n, p):
    front = int(p/2)
    if n % 2 == 1:
        back = int(((n-p)/2))
    else:
        back = int(((n-p+1)/2))

    return min(front, back)

File: JS_Problems/Drawing_Book.js
/*
 * Complete the 'pageCount' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. INTEGER p
 */

function pageCount(n, p) {
  const front = Math.floor(p / 2);
  let back;
  if (n % 2 === 1) {
    back = Math.floor((n - p) / 2);
  } else {
    back = Math.floor((n - p + 1) / 2);
  }
  return Math.min(front, back);
}
